main/e2e-testing.md

```diff
@@ -0,0 +1,145 @@
+        └── specs
+            └── test.spec.js
+```
+
+## Configure `synpress.config.js`
+
+Before you start writing your E2E tests, you'll need to create a configuration file named `synpress.config.js` in the `tests/e2e` directory of your project.
+
+Inside `synpress.config.js`, you'll set the `baseUrl` property within the e2e configuration. This tells Cypress where to find your DApp during testing. Here's an example configuration:
+
+```js
+const baseConfig = require('@agoric/synpress/synpress.config');
+const { defineConfig } = require('cypress');
+
+module.exports = defineConfig({
+  ...baseConfig,
+  e2e: {
+    ...baseConfig.e2e,
+    baseUrl: 'http://localhost:5173',
+  },
+});
+```
+
+In this example, `baseUrl` is set to `http://localhost:5173`. Make sure to replace this with the actual URL where your DApp is running.
+
+By default, if you don't create a `synpress.config.js` file, `@agoric/synpress` assumes your DApp runs on port 3000 of your local machine (`http://localhost:3000`). Specifying the correct `baseUrl` ensures your tests interact with the right instance of your DApp.
+
+Since you have a separate `synpress.config.js` in the `tests/e2e` folder, you need to specify its location when running your tests. Use the `--configFile` flag with the `synpress run` command:
+
+```bash
+EXTENSION=keplr synpress run --configFile=tests/e2e/synpress.config.js
+```
+
+## Creating a Support File
+
+Navigate to your project's `tests/e2e` directory and create a new file named `support.js`. This file can be used to create reusable commands that encapsulate common testing actions. For example, you can create a custom command to navigate to a specific page within your application. You can read more about it [over here](https://docs.cypress.io/api/cypress-api/custom-commands).
+
+After creating your `support.js` file, make sure to include the following import statement:
+
+```js
+import '@agoric/synpress/support/index';
+```
+
+This import is essential because it brings in the necessary functionalities from `@agoric/synpress` to interact with the Keplr wallet within your e2e tests. Without it, your tests won't be able to leverage the features provided by `@agoric/synpress` for Keplr integration.
+
+## Example Tests
+
+With the environment set up, let's write end-to-end (e2e) tests to test your DApp's functionality. In the `tests/e2e/specs` folder, create a new file with a descriptive name that reflects what it tests (e.g., `user_login.spec.js` or `token_transfer.spec.js`).
+
+### Test Structure
+
+You use `describe` blocks to group related tests together, and `it` blocks to define individual test cases within those groups.
+
+```js
+describe('User Login', () => {
+  it('should login with valid credentials', () => {
+    // Test steps for login functionality
+  });
+});
+```
+
+### Test 1: Setting Up Keplr Wallet
+
+```js
+it('should setup a Keplr wallet', () => {
+  cy.setupWallet({
+    secretWords: 'REPLACE_WITH_YOUR_KEPLR_MNEMONIC',
```

Comment:
Can you shorten `REPLACE_WITH_YOUR_KEPLR_MNEMONIC` to `KEPLR_MNEMONIC`, because you are already explaining to replace it in the paragraph bellow.